mypy/indirection.py

```diff
@@ -0,0 +1,101 @@
+from typing import Optional, Set, Iterable
+from abc import abstractmethod
+
+from mypy.visitor import NodeVisitor
+from mypy.types import TypeVisitor
+from mypy.nodes import MODULE_REF
+import mypy.nodes as nodes
+import mypy.types as types
+from mypy.util import split_module_names
+
+
+def extract_module_names(type_name: Optional[str]) -> Iterable[str]:
+    """Returns the module name of a fully qualified type name."""
```

Comment:
Hm, it really yields a sequence of module names. Also, couldn't this be implemented by calling util.split_module_names() on the type name less the last segment?